packages/block-library/src/button/edit.js

```diff
@@ -293,12 +293,23 @@ function ButtonEdit( props ) {
 						...spacingProps.style,
 						...shadowProps.style,
 					} }
-					onSplit={ ( value ) =>
-						createBlock( 'core/button', {
-							...attributes,
-							text: value,
-						} )
-					}
+					onSplit={ ( value, isOriginal ) => {
+						let newAttributes;
+						if ( isOriginal || value ) {
+							newAttributes = {
+								...attributes,
+								text: value,
+							};
+						}
+						const block = createBlock(
+							'core/button',
+							newAttributes
+						);
+						if ( isOriginal ) {
+							block.clientId = clientId;
```

Comment:
Why is this necessary?

Reply:
I assumed it is needed because when we split a paragraph or a button we are actually creating two new blocks, and I guess that [createBlock](https://github.com/WordPress/gutenberg/blob/df1730c0d4b28c9f47daea399e413be76189c7c5/packages/block-library/src/paragraph/edit.js#L162-L163) generates a new `clientId`. However, we might want to keep the original clientId to simulate that it is actually still the original block.